porting scheduler for Avr

vbextreme opened the pull request:
porting scheduler for Arduino One, for now no other board with avr chip.

Testato committed:
Create MultiplePWM.ino

new Example

Testato committed:
Create Loop2.ino

New Example

Testato committed:
Create Loop3.ino

new example

Testato committed:
Update MultiplePWM.ino

Testato committed:
Update Loop2.ino

Testato committed:
Update Loop3.ino

Testato committed:
Update MultiplePWM.ino

Testato commented:
+1

facchinm commented:
Great effort @vbextreme, I'll test it as soon as I can. Is there any reason why it only works on 328 or is the code suitable for every 8bit AVR chip (like Mega2560 and 32u4)?

Testato committed:
New MultipleSketch example

Testato committed:
Create Sketch2

Testato committed:
Create Sketch3

Testato committed:
Update keywords.txt

vbextreme commented:
first I need board mega ad leonardo.
second need to study a mega && 32u4 datasheet.
and the last need to implement, is not very simple.
for now wait confirm is ok

vbextreme committed:
porting to all avr chip

porting to all avr chip

vbextreme committed:
Merge pull request #1 from Testato/patch-1

New examples

vbextreme commented:
Togle error gcc for 22bit addressing and add some example

vbextreme committed:
add avr in library.properties

fix malloc(It will be strengthened as soon as possible), fix interrupt, need clean code

prepared sizing loop , remove conflict malloc , insert stack min

use standard malloc and resolve bug, add start stop task, priority, semaphore, mutex and example

add optimization possibilities for cooperative scheduler , to be enabled in scheduler.h

errata corrige

Testato commented:
News on this pr ?
It will be merged ?

q2dg commented:
Please...this would be a great step forward

vbextreme commented:
ported to the repository is not complete , I worked for a new version soon as I find some time I finish the job started .

asukiaaa commented:
The following scheduler allows me to compile multiple blink for leonardo but not for MEGA.
Does this help you?

https://github.com/per1234/SCoop/blob/master/SchedulerARMAVR/SchedulerARMAVR.cpp

Thank you.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/arduino-libraries/Scheduler?pullRequest=1) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/arduino-libraries/Scheduler?pullRequest=1) before we can accept your contribution.<br/>**0** out of **2** committers have signed the CLA.<br/><br/>:x: vbextreme<br/>:x: Testato<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/arduino-libraries/Scheduler?pullRequest=1) it.</sub>

Letsric commented:
Any updates? This is still an awesome library, and the merge conflicts with `master` of `arduino-libraries/Scheduler` seem easy to resolve. Here is an example with the merged version of the lib on Wokwi: <https://wokwi.com/projects/434992178008162305>
